drivers/axi_core/axi_dac_core/axi_dac_core.c

```diff
@@ -363,6 +385,206 @@ int32_t axi_dac_write(struct axi_dac *dac,
+{
+	uint32_t reg;
+
+	no_os_axi_io_read(dac->base, reg_addr, &reg);
+
+	reg &= ~reg_mask;
+	reg |= reg_data;
+
+	no_os_axi_io_write(dac->base, reg_addr, reg);
+}
+
+/**
+ * @brief AXI DAC Read Data until condition is met.
+ * @param dac - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - The data mask to apply.
+ * @param value - The data value to check.
+ * @param sleep_us - Poll interval.
+ * @param timeout_us - Poll timepout.
+ */
+int32_t axi_dac_read_poll_timeout(struct axi_dac *dac, uint32_t reg_addr,
+				  uint32_t mask, uint32_t value,
+				  uint32_t sleep_us, uint32_t timeout_us)
+{
+	uint32_t reg;
+
+	while (timeout_us) {
+		no_os_axi_io_read(dac->base, reg_addr, &reg);
+
+		if ((reg & mask) == value)
+			return 0;
+
+		no_os_udelay(sleep_us);
+		timeout_us -= sleep_us;
+	}
+
+	return -ETIMEDOUT;
+}
+
+/**
+ * @brief AXI DAC Bus Data Write.
+ * @param dac - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - Data value to be written.
+ * @param data_size - Data size in bytes.
+ * @return Returns 0 in case of success or negative error code otherwise.
+ */
+int32_t axi_dac_bus_write(struct axi_dac *dac, uint32_t reg_addr,
+			  uint32_t reg_data, uint8_t data_size)
+{
+	int err;
+
+	if (dac->bus_type != AXI_DAC_BUS_TYPE_QSPI)
+		return -EINVAL;
+
+	no_os_axi_io_write(dac->base, AXI_DAC_CNTRL_DATA_WR,
+			   (data_size == 2) ?
+			   AXI_DAC_DATA_WR_16(reg_data) :
+			   AXI_DAC_DATA_WR_8(reg_data));
+
+	axi_dac_update_bits(dac, AXI_DAC_REG_CNTRL_2, AXI_DAC_SYMB_8B,
+			    (data_size == 1) ? AXI_DAC_SYMB_8B : 0);
+
+	axi_dac_update_bits(dac, AXI_DAC_REG_CUSTOM_CTRL,
+			    AXI_DAC_ADDRESS(0xff), AXI_DAC_ADDRESS(reg_addr));
+
+	axi_dac_update_bits(dac, AXI_DAC_REG_CUSTOM_CTRL,
+			    AXI_DAC_TRANSFER_DATA, AXI_DAC_TRANSFER_DATA);
+
+	err = axi_dac_read_poll_timeout(dac, AXI_DAC_REG_CUSTOM_CTRL,
+					AXI_DAC_TRANSFER_DATA,
+					AXI_DAC_TRANSFER_DATA, 10, 100000);
+	if (err) {
+		printf("bus read poll timeout error\n");
+		return err;
+	}
+
+	no_os_udelay(100);
+
+	axi_dac_update_bits(dac, AXI_DAC_REG_CUSTOM_CTRL,
+			    AXI_DAC_TRANSFER_DATA, 0);
+
+	return 0;
+}
+
+/**
+ * @brief AXI DAC Bus Data Read.
+ * @param dac - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_data - Pointer to data value to be read.
+ * @param data_size - Data size in bytes.
+ * @return Returns 0 in case of success or negative error code otherwise.
+ */
+int32_t axi_dac_bus_read(struct axi_dac *dac,
+			 uint32_t reg_addr,
+			 uint32_t *reg_data,
+			 uint8_t data_size)
+{
+	int err;
+
+	if (dac->bus_type != AXI_DAC_BUS_TYPE_QSPI)
+		return -EINVAL;
+
+	err = axi_dac_bus_write(dac, AXI_DAC_RD_ADDR(reg_addr), 0, data_size);
+	if (err) {
+		printf("bus write error\n");
+		return err;
+	}
+	err = axi_dac_read_poll_timeout(dac, AXI_DAC_UI_STATUS,
+					AXI_DAC_BUSY, 0, 10, 100);
+	if (err) {
+		printf("bus timeout error\n");
```

Comment:
please remove all these printf statements from the driver, this will mess the iio communication over UART

Reply:
done